### ISSUE TITLE
Live reloading for Server Side Blazor

### ISSUE THREAD

**Issue by BickelLukas** [aspnet/Blazor#1704]
Is live reloading supported in server side blazor yet? 

It should work just like it does with regular razor views in AspNetCore.

**Comment by SteveSandersonMS** [comment]
Yes, once you edit a `.cshtml` file, next time you reload the page in your browser it will recompile in VS before the page reloads. Note that this is currently only supported if you're using VS (not VS Code).

**Comment by BickelLukas** [comment]
@stevesandersonms This only works for me when using the regular webassebly version (Either Hosted or Standalone)

However it didn't work when using Server Side Blazor (Neither when debugging, nor when launching the app with CTRL+F5). Do I need to do any additional configuration to get it that work?

**Comment by cores-system** [comment]
@SteveSandersonMS, I have the same problem.

**Comment by danroth27** [comment]
@BickelLukas @cores-system I'm not able to reproduce this. Could you please provide more specific details on how to reproduce the issue?:
- Blazor version?
- VS version?
- .NET Core version?
- Repo project or specific steps to create the project?
- Specific steps followed, observed & expected behavior?

**Comment by BickelLukas** [comment]
@danroth27 

just updated everything to the latest Version.

Blazor version: 0.7.0
VS version: 15.8.9
.NET Core version: 2.1.500
Blazor Extension Version: 15.7.20181115.1

Steps followed: 
1. File -> New Project -> Server Side Blazor
2. Start with CTRL + F5
3. Change content in cshtml file
4. Reload Browser

Expected Behavior: 
Changes are shown in the Browser.

Observed Behavior:
The Browser is still showing the state that was present when the project was launched

Heres a screen recording of exactly what I did.
[Screen Recording.zip](https://github.com/aspnet/Blazor/files/2621841/Screen.Recording.zip)

**Comment by michaelvolz** [comment]
To integrate watch and live reloading you can add this to your `launchsettings.json`:
```
    "Watch": {
      "executablePath": "dotnet.exe",
      "commandLineArgs": "watch --project ..\\..\\..\\..\\YOURPROJECT.csproj run -c Debug",
      "launchBrowser": true,
      "launchUrl": "http://localhost:5000/",
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      }
    }
```

**Comment by dharmaturtle** [comment]
To be clear, that edit should be made inside `profiles`. You can also avoid the `--project...` arg if you use `"workingDirectory": "$(ProjectDir)",`, like so:

    "dotnet watch run": {
      "commandName": "Executable",
      "executablePath": "dotnet",
      "workingDirectory": "$(ProjectDir)",
      "commandLineArgs": "watch run",
      "launchBrowser": true,
      "launchUrl": "http://localhost:5000/",
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      }
    }

However, I don't have success with the browser opening despite the above. Does anyone else have success with that?

**Comment by michaelvolz** [comment]
Thanks for the update. But you have to open the browser manually as far as I know.

**Comment by WhistlerHusky** [comment]
Live reloading works with two settings.

1. @dharmaturtle said.
2.  `  <ItemGroup>
      <Watch Include="..\**\*.razor" />
  </ItemGroup>`

Now I wonder if there is a way to force to refresh the browser after saving code changed.
This feature works with Angular. so I don't have to refresh the browser.

**Comment by dharmaturtle** [comment]
@WhistlerHusky yes, you can make the browser refresh if you change the code. https://github.com/martasp/BlazorLiveReload
